### DISCUSSION TITLE
Entity Linker Returning NER label instead of KB label

### DISCUSSION THREAD

**Discussion by ianteohsc** [explosion/spaCy#9569] [Help: Coding & Implementations] [answered]
I've just trained the Entity Linker and am now testing it against unseen text. I noticed that the ent.label_ property is showing my NER label output instead of WikiData's Q-ids. 

```
doc1_text="..."

doc = nlp(doc1_text)

for ent in doc.ents:
    print(ent.text, ent.label_, ent.kb_id_, ent.kb_id, ent.label)

#Console output
'''
macular dystrophy DIAGNOSIS NIL 8903711302469904182 7573096171271069708
Stargardts DIAGNOSIS NIL 8903711302469904182 7573096171271069708
Avastin MEDICARE NIL 8903711302469904182 1725762881541326455
SRNVM DIAGNOSIS NIL 8903711302469904182 7573096171271069708
macular dystrophy DIAGNOSIS NIL 8903711302469904182 7573096171271069708
'''
```

I've prepared the training data as per the Entity Linker's jupyter notebook example. Here is the output of the first row of my training data.

```
dataset[0][1]
{'entities': [(207, 215, 'DIAGNOSIS')], 'links': {(207, 215): {'Q127724': 1.0}}}
```

I've tried swapping the "DIAGNOSIS" in the entities' array's tuple to Q-id and the result is the same.

Have i misconfigured my training data or should i look else where to address this ? Any help will be greatly appreciated.

**Comment by polm** [comment, marked as answer]
> I noticed that the ent.label_ property is showing my NER label output instead of WikiData's Q-ids.

This is normal. If you look at the provided evaluate script, you'll notice that the Q IDs are in the `kb_id_` field, not the `label_` field. It looks like all your `kb_id_`s are `NIL`, which is a separate problem.

**Reply by ianteohsc** [reply]
Thank you for the confirmation. This community is awesome.
